vars/valgrindReportPublish.groovy

```diff
@@ -30,21 +30,25 @@ def call(Map config = [:]) {
     // Need to have only one Valgrind publish stage
     println "No valgrind_stashes passed!   Running older code!"
   }
+  
+  sh label: 'Remove *.memcheck.xml files in this workspace',
+     script: '''rm -rf *.memcheck.xml'''
```

Comment:
You don't need triple quotes here.

```suggestion
     script: 'rm -rf *.memcheck.xml'
```
But why `-r` for what looks like it should be a file pattern?  And why not use `fileOperations->fileDeleteOperation` instead of a `sh` step here?

Reply:
I use the -r because of the dirty workspaces problem it may be the case these files do not exist (it they're not here from previous builds). In fact in my experiments sometimes I saw memcheck result files with very old names instead of the *memcheck.xml pattern here. I don't know how fileOperations will behave in the case of a workspace with no files that match this pattern

Reply:
I'm going to send a patch deleting a file that do not exist to see how this behaves, if it doesn't throw an error, I think we can go safe with fileOperations

Reply:
-f handles the non-existent case.  -r is for recursion which we probably don't want.

Reply:
Oh, right, my mistake. 

I've addressed Brian's comments now. I removed the triple quote and use fileDeleteOperation instead of rm.

Reply:
@marcelarosalesj You should leave conversations to be resolved by the person who started them, so that they can see what their previous issues were and whether they were resolved or not.